sound/soc/sof/core.c

```diff
@@ -282,13 +294,22 @@ static int sof_probe(struct platform_device *pdev)
 		goto dbg_err;
 	}
 
+	/*
+	 * let's register miscdev now.
+	 * This will be used by the device manager in hypervisor
+	 * so that vFE can communicate with vBE
+	 */
+	sof_virtio_miscdev_register(sdev);
+
 	/* init the IPC */
 	sdev->ipc = snd_sof_ipc_init(sdev);
 	if (!sdev->ipc) {
 		dev_err(sdev->dev, "error: failed to init DSP IPC %d\n", ret);
 		goto ipc_err;
 	}
 
+	/* vFE will not touch HW. Let's skip fw loading */
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_VIRTIO_FE)
```

Comment:
As suggested on V3, lets use a runtime sdev->is_vfe or sdev->is_vbe to select this code at runtime. These states will be set at probe() time.

Reply:
Although we can use such flag to judge whether it is vFE or not, we still need such '#ifdef' because acrn has compile conflict as you know. We have already talked about that before. vFE and vBE must use such '#ifdef' to avoid the compiling error. After acrn has fixed these issues, I will switch to flag and remove the '#ifdef'

Reply:
@libinyang the point I'm making is that we only use the ifdef in as few places as possible. To minimize the contamination.